src/core/sensor/qgssensormanager.h

```diff
@@ -0,0 +1,127 @@
+/***************************************************************************
+    qgssensormanager.h
+    ------------------
+    Date                 : March 2023
+    Copyright            : (C) 2023 Mathieu Pellerin
+    Email                : mathieu at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSSENSORMANAGER_H
+#define QGSSENSORMANAGER_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgsabstractsensor.h"
+
+#include <QObject>
+#include <QDomElement>
+
+class QgsProject;
+
+/**
+ * \ingroup core
+ * \class QgsSensorManager
+ *
+ * \brief Manages sensors.
+ *
+ * QgsSensorManager handles the storage, serializing and deserializing
+ * of sensors. Usually this class is not constructed directly, but
+ * rather accessed through a QgsProject via QgsProject::sensorManager().
+ *
+ * \since QGIS 3.32
+ */
+class CORE_EXPORT QgsSensorManager : public QObject
+{
+    Q_OBJECT
+
+  public:
+
+    /**
+     * Returns a newly created QgsSensorManager linked to the specified \a project.
+     *
+     * The returned object is parented to the \a project.
+     */
+    static QgsSensorManager *createProjectBasedManager( QgsProject *project );
```

Comment:
Why not just make QgsSensorManager constructor require a QgsProject parent argument?

Reply:
@nyalldawson , because we'd want people to be allow to have non-project bound sensor managers. But I realize here we don't actually make use of the project variable yet, so I can remove that altogether for now.